### PR TITLE
index broadcasttxinfo correctly on rbf

### PR DESCRIPTION
**Pull request by tomt1664**

### CODE REVIEW

File: src/wallet/wallet.ts [hunk 1]
```diff
@@ -1559,7 +1559,7 @@ export class Wallet {
       if (fee_max > 0) {
         if (fee_max >= fee_per_byte) throw Error(`Requested fee per byte ${fee_per_byte} is not greater than existing fee per byte ${fee_max}`);
         const ids_sorted_1 = shared_key_ids.slice().sort();
-        const ids_sorted_2 = broadcastTxInfos[0].withdraw_msg_2.shared_key_ids.slice().sort();
+        const ids_sorted_2 = broadcastTxInfos[broadcastTxInfos.length - 1].withdraw_msg_2.shared_key_ids.slice().sort();
```
Comment (lawlawlaw) [inline src/wallet/wallet.ts:1562]: Does it make any difference? This value should be the same for all items of broadcastTxInfos.

**Review by lawlawlaw — CHANGES_REQUESTED** [review]

File: src/wallet/wallet.ts (hunk 1 above)
Comment (tomt1664) [reply src/wallet/wallet.ts:1562]: Normally, yes. But if the wallet is recovered with withdrawing coins, an empty/dummy `tx_withdraw_broadcast` element is added as the previous broadcast tx info is lost/unavailable. 

This fixes the error you were getting when trying to re-re-sign a recovered withdrawing coin.

Comment (lawlawlaw) [reply src/wallet/wallet.ts:1562]: Does line 1575 also need to be changed as this also accesses the first element?

**Review by lawlawlaw — APPROVED** [review]

**Review by lawlawlaw — CHANGES_REQUESTED** [review]
When starting the wallet in testnet mode, it uses the mainnet server address.

**Comment by lawlawlaw** [comment]
> When starting the wallet in testnet mode, it uses the mainnet server address.



> When starting the wallet in testnet mode, it uses the mainnet server address.

Fixed here: https://github.com/layer2tech/mercury-wallet/pull/1360

**Pull request merged**